util/pbtypes/pbtypes.go

```diff
@@ -582,3 +606,46 @@ func RelationIdToKey(id string) (string, error) {
 
 	return "", fmt.Errorf("incorrect id format")
 }
+
+// AddValue adds values to int lists and string lists
+func AddValue(s *types.Struct, key string, v *types.Value) {
```

Comment:
I don't think these functions should be here, they are pretty specialized for only one task, so maybe move them close to place when they're called?

Reply:
I moved it to separate file core/block/details.go, so all block.Service operations on details are described here